### PR TITLE
Us012 add new job role

### PR DESCRIPTION
**Pull request by mickey-dawson**
Added create job webpage with service and controller

### CODE REVIEW

File: controller/createJobController.ts [hunk 1]
```diff
@@ -0,0 +1,29 @@
+import { createIncrementalCompilerHost } from 'typescript';
+import { createJob } from '../model/createJob';
+import { Application, Request, Response } from "express";
+const createJobService = require('../service/createJobService')
+
+module.exports = function(app: Application){
+
+    app.get('/create-job-roles', async (req: Request, res: Response) =>{
+        res.render('create-job-roles')
```
Comment (shaunganley) [inline controller/createJobController.ts:9]: you're not getting the bands and capabilities from the API to populate the drop-down lists

File: service/createJobService.ts [hunk 2]
```diff
@@ -0,0 +1,15 @@
+import { createJob } from '../model/createJob';
+const axios = require('axios');
+
+module.exports.createJob = async function (Job : createJob ) {
+    try {
+        const response = await axios.post(process.env.API_URL+'/api/create-job-roles');
```
Comment (shaunganley) [inline service/createJobService.ts]:
```suggestion
        const response = await axios.post(process.env.API_URL+'/api/create-job-roles', createJob);
```

**Review by shaunganley — CHANGES_REQUESTED** [review]
you need to get the bands & capabilities from the API rather than hardcoding in the view. you also need unit tests

File: controller/createJobController.ts (hunk 1 above)
Comment (mickey-dawson) [reply controller/createJobController.ts:9]: > you're not getting the bands and capabilities from the API to populate the drop-down lists

The form in views  that encapsulates those lists makes a post request to the database so they should be connected to the database

File: .github/test/service/unit [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+
+        try {
+          await createJobService.createJob()
+        } catch (e) { 
+          var error = e.message
+        }
+        expect(error).to.equal('Could not create JobRoles')
+      })
+    })
+})
```
Comment (glenn-hamilton) [inline .github/test/service/unit]: New lines on the end of every file

File: .DS_Store
Comment (glenn-hamilton) [inline .DS_Store:1]: Why?

Comment (mickey-dawson) [reply .DS_Store:1]: why not?

File: .github/test/service/unit
Comment (shaunganley) [inline .github/test/service/unit:1]: please move this to the correct folder

**Review by shaunganley — APPROVED** [review]

**Review by glenn-hamilton — APPROVED** [review]

**Pull request merged**